linker/shared/src/main/scala/org/scalajs/linker/backend/wasmemitter/ClassEmitter.scala

```diff
@@ -785,24 +796,46 @@ class ClassEmitter(coreSpec: CoreSpec) {
    */
   private def genGlobalClassItable(clazz: LinkedClass)(implicit ctx: WasmContext): Unit = {
     val className = clazz.className
-
-    if (ctx.getClassInfo(className).classImplementsAnyInterface) {
-      val globalID = genGlobalID.forITable(className)
-      val itablesInit = List(
-        wa.I32Const(ctx.itablesLength),
-        wa.ArrayNewDefault(genTypeID.itables)
-      )
-      val global = wamod.Global(
-        globalID,
-        makeDebugName(ns.ITable, className),
-        isMutable = false,
-        watpe.RefType(genTypeID.itables),
-        wa.Expr(itablesInit)
+    val classInfo = ctx.getClassInfo(className)
+    if (classInfo.classImplementsAnyInterface) {
+      genGlobalClassItable(
+        genGlobalID.forITable(className),
+        classInfo,
+        clazz.ancestors
       )
-      ctx.addGlobal(global)
     }
   }
 
+  private def genGlobalClassItable(classITableGlobalID: wanme.GlobalID,
+      classInfoForResolving: WasmContext.ClassInfo, ancestors: List[ClassName])(
+      implicit ctx: WasmContext): Unit = {
+    val itablesInit = Array.fill[List[wa.Instr]](ctx.itablesLength) {
+      List(wa.RefNull(watpe.HeapType.Struct))
+    }
+    val resolvedMethodInfos = classInfoForResolving.resolvedMethodInfos
+
+    for {
+      ancestor <- ancestors
+      // Use getClassInfoOption in case the reachability analysis got rid of those interfaces
+      interfaceInfo <- ctx.getClassInfoOption(ancestor)
+      if interfaceInfo.isInterface
+    } {
+      val init = interfaceInfo.tableEntries.map { method =>
+        ctx.refFuncWithDeclaration(resolvedMethodInfos(method).tableEntryID)
```

Comment:
Since this is now in a `global` section, we don't need separate declarations anymore:
```suggestion
        wa.RefFunc(resolvedMethodInfos(method).tableEntryID)
```